File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int result = 0;
        unordered_map<int, int> map;
        for (int num : nums) {
            map[num]++;
        }

        for (auto const& [key, count] : map) {
            if (map.count(key + 1)) {
                result = std::max(result, count + map[key + 1]);
            }
        }
        return result;
    }
};

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.py
class Solution(object):
    def findLHS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        counts = Counter(nums)

        for key in counts:
            if (key + 1) in counts:
                result = max(result, counts[key] + counts[key + 1])
                
        return result